### PR TITLE
FLPATH-418: Add endpoints and tasks for project access request workflow

### PR DESCRIPTION
**Pull request by anludke**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (FLPATH-xxxx)*:
Fixes #

**Change type**
- [x] New feature
- [ ] Bug fix
- [x] Unit tests
- [ ] Integration tests
- [ ] CI
- [ ] Documentation
- [x] Auto-generated SDK code

**Impacted services**
- [x] Workflow Service
- [ ] Notification Service

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+			accessRequestId = UUID.fromString(getRequiredParameterValue(ACCESS_REQUEST_ID));
+		}
+		catch (MissingParameterException e) {
+			log.error("Exception when trying to get required parameter(s): {}", e.getMessage());
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		try {
+			String url = String.format("http://%s:%s/api/v1/projects/access/%s", serviceUrl, servicePort,
+					accessRequestId);
```
Comment (masayag) [inline prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java:78]: What is the service/server that should reply to `"http://%s:%s/api/v1/projects/%s/access` ?

Comment (masayag) [reply prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java:78]: So this is the new project access endpoint:
```
@RequestMapping("/api/v1/projects/access")
```
why not use rest call? shouldn't we use some service API call or is the call done via HTTP to remain detached from the service?

Comment (anludke) [reply prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java:81]: > What is the service/server that should reply to `"http://%s:%s/api/v1/projects/%s/access` ?

The task uses the notification service to send this approval url to the approver.

Comment (anludke) [reply prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java:81]: > 
Actually, the url is the endpoint path to make a POST HTTP call to approve the request. Ideally there will be a UI with the form to approve/reject the request with a comment.
The task showcases how the endpoint and required params (access request id) will send out to the approver via notification service.

File: workflow-service/src/main/resources/db/changelog/tables.xml [hunk 2]
```diff
@@ -72,6 +72,32 @@
...
+            <column name="project_id" type="${uuidType}">
+                <constraints nullable="false" foreignKeyName="fk_project_access_request_project" references="prds_project(id)"/>
+            </column>
+            <column name="user_id" type="${uuidType}">
+                <constraints nullable="false" foreignKeyName="fk_project_access_request_USER" references="prds_user(id)"/>
+            </column>
+            <column name="role_id" type="${uuidType}">
+                <constraints nullable="false" foreignKeyName="fk_project_access_request_role" references="prds_role(id)"/>
+            </column>
+            <column name="status" type="varchar(255)"/>
```
[the remark this answers is not in the document]
Comment (anludke) [reply workflow-service/src/main/resources/db/changelog/tables.xml]: Fixed

File: prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java (hunk 1 above)
Comment (masayag) [reply prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java:81]: I'd rather have this implemented in a similar fashion as https://github.com/parodos-dev/parodos/blob/main/workflow-service/src/main/java/com/redhat/parodos/NotificationSender.java#L21
Leveraging the common SDK to interact with the service endpoint.

Comment (anludke) [reply prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/project/checker/ProjectAccessRequestApprovalWorkFlowCheckerTask.java:81]: I was debating on using the SDK but @RichardW98 convinced me :)
I'm happy to refactor this in a separate PR. I'll create a tech debt to track and address this.
Thanks

**Pull request merged**